src/diffr_lib/mod.rs

```diff
@@ -663,25 +663,29 @@ pub fn tokenize(src: &[u8], ofs: usize, tokens: &mut Vec<Span>) {
             tokens.push((lo, hi))
         }
     };
+    let mut kind = GraphemeKind::Other;
     let mut lo = ofs;
-    let mut kind = TokenKind::Other;
     #[allow(clippy::needless_range_loop)]
-    for hi in ofs..src.len() {
+    for (grapheme_start, _, g) in src[ofs..].grapheme_indices() {
+        let hi = grapheme_start + ofs;
         let oldkind = kind;
-        kind = classify_byte(src[hi]);
-        if kind != oldkind || oldkind == TokenKind::Other {
+        kind = classify_grapheme(g);
+        if kind != oldkind || oldkind == GraphemeKind::Other {
             push(lo, hi);
             lo = hi
         }
     }
     push(lo, src.len());
 }
 
-fn classify_byte(b: u8) -> TokenKind {
-    match b {
-        b'a'..=b'z' | b'A'..=b'Z' | b'0'..=b'9' | b'_' => TokenKind::Word,
-        b'\t' | b' ' => TokenKind::Spaces,
-        _ => TokenKind::Other,
+fn classify_grapheme(g: &str) -> GraphemeKind {
+    let first_char = g.chars().next().unwrap_or_default();
+    if first_char.is_alphanumeric() {
```

Comment:
This introduces a change, since we now can break in the middle of an identifier with '_' in the name.
The motivation for the original code was to keep those identifiers intact for languages with snake-case convention (like rust!).

Example input file impacted by the change:
```
@@ -383,7 +384,7 @@ pub fn parse_config() -> AppConfig {
     let mut args = args().peekable();
     while parse_options(&mut config, &mut args) {}
 
-    if atty::is(atty::Stream::Stdin) {
+    if std::io::stdin().is_terminal() {
         usage(-1);
```

<img width="654" alt="image" src="https://github.com/mookid/diffr/assets/7757342/e97a69f4-f5a6-4af6-811d-e7dca0ac0647">